File: src/main/java/com/example/SimpleHttpServer.java
package com.example;

import com.sun.net.httpserver.HttpServer;
import com.sun.net.httpserver.HttpHandler;
import com.sun.net.httpserver.HttpExchange;

import java.io.IOException;
import java.io.OutputStream;
import java.net.InetSocketAddress;

public class SimpleHttpServer {

    public static void main(String[] args) throws IOException {
        // Create an HttpServer instance
        HttpServer server = HttpServer.create(new InetSocketAddress(8080), 0);

        // Create a context for the root path and set the handler
        server.createContext("/", new HttpHandler() {
            @Override
            public void handle(HttpExchange exchange) throws IOException {
                String response = "Hello, World!";
                exchange.sendResponseHeaders(200, response.getBytes().length);
                try (OutputStream os = exchange.getResponseBody()) {
                    os.write(response.getBytes());
                }
            }
        });

        // Start the server
        server.start();
        System.out.println("Server is listening on port 8080...");
    }
}

File: src/main/kotlin/com/example/SimpleHttpServer.kt
package com.example

import com.sun.net.httpserver.HttpExchange
import com.sun.net.httpserver.HttpHandler
import com.sun.net.httpserver.HttpServer
import java.io.IOException
import java.net.InetSocketAddress

object SimpleHttpServer {
    @JvmStatic
    @Throws(IOException::class)
    fun main(args: Array<String>) {
        val server: HttpServer = HttpServer.create(InetSocketAddress(8080), 0)

        server.createContext("/", HttpHandler { exchange ->
            val response = "Hello, World!"
            exchange.sendResponseHeaders(200, response.toByteArray().size.toLong())
            exchange.responseBody.use { os ->
                os.write(response.toByteArray())
            }
        })

        server.start()
        println("Server is listening on port 8080...")
    }
}
